### ISSUE TITLE
[sklearn#33584] Adding array API support to `sklearn.covariance`

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#388]
**Mirrored from [scikit-learn#33584](https://github.com/scikit-learn/scikit-learn/issues/33584)**

Original title: Adding array API support to `sklearn.covariance`
Original labels: Array API
Created: 2026-03-19T15:43:07Z

---

This is a mega-issue tracking work on adding array API support to the parts of `sklearn.covariance` where it makes sense.

> [!IMPORTANT]
>  This is a fresh issue that still needs refining before it is ready for contribution. Please don't yet open PRs.

This is a list of all the items in the covariance module, they are sorted by (roughly) the order in which they should be tackled.

- [ ] `LedoitWolf` https://github.com/scikit-learn/scikit-learn/pull/33573
- [ ] `OAS` https://github.com/scikit-learn/scikit-learn/pull/33600
- [ ] `ShrunkCovariance`
- [ ] `GraphicalLasso`
- [ ] `GraphicalLassoCV`
- [ ] `MinCovDet`
- [ ] `EllipticEnvelope`
- [ ] `EmpiricalCovariance` (tackle this last, check what can be refactored from the classes inheriting from it into this base class

Functions listed here should be tackled with the class (from above):
- [ ] `empirical_covariance`
- [ ] `graphical_lasso`
- [ ] `ledoit_wolf` https://github.com/scikit-learn/scikit-learn/pull/33573
- [ ] `ledoit_wolf_shrinkage`
- [ ] `oas`
- [ ] `shrunk_covariance`

Please refer to this issue when opening a PR to tackle something from this list. Use `Towards #33584` to refer to this issue, do not use "closes" or "fixes" (that would close this issue when merging your PR).

After the `LedoitWolf` and `OAS` PRs are merged we will consider opening this issue for a wider audience of contributors.

~~The next thing to work out is in which order items should be tackled. This will require looking at the existing PR for `LedoitWolf` and investigating how the other estimator classes depend on each other. If you investigate this and find out an order that makes sense please post in this issue.~~

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [7](https://github.com/godofecht/flow-scikit/issues/7) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
